### ISSUE TITLE
Social login implementations

### ISSUE THREAD

**Issue by santanaruben** [jon-sully/react-netlify-identity-gotrue#3]
Hi @jon-sully, great job, I wanted to ask if there is any way to implement social logins. In the [gotrue-js](https://github.com/netlify/gotrue-js#oauth-providers-supported-by-netlify) repo they explain that currently, they 

> support Google, GitHub, GitLab, and BitBucket directly in the UI (other OAuth providers require serverless functions to be configured correctly, but these do not). 

And in the [netlify-identity-widget demo](https://identity.netlify.com/) it is possible to use them.
Is there any process I can develop to achieve this using this package?

**Comment by santanaruben** [comment]
?

**Comment by jon-sully** [comment]
Greetings @santanaruben! And apologies on not catching your issue sooner — I had to reconfigure some GitHub notification settings and missed it. D'oh!

At this point I'm not necessarily rushing to implement social / SSO login options. I promise it's not because I'm a curmudgeon or want to stand up against Google etc. etc. — it's actually because it's dangerous with how some of the Netlify Identity plumbing works and can expose folks to gaping security holes they don't mean to have (like how _anybody_ can log into your Netlify CMS if you enable social logins.. whoops!). I explained a little more over on the [Netlify Forums](https://answers.netlify.com/t/identity-weird-state/32187/11?u=jonsully):

> I’ll certainly give it a serious look-over, but the issues that come from the SSO stuff don’t have anything to do with the implementation :sweat_smile: moreso just the side-effects. E.g. if you have SSO from Google enabled, even if you have Netlify Identity set to “by invite only”, I’m pretty sure SSO circumvents that. And identity event-based Functions don’t always work the way you’d always think with the SSO providers, etc.
> 
> So it’s not so much that I don’t want to allow folks to use SSO with the library, it’s actually out of caution for other side-effects that can really cause some security concerns.

If I wanted to enable this library to provide bindings for SSO-style logins I'd also need to provide significant amounts of documentation that outlines all of the security issues that can arise and how to correctly configure everything.. which is both a large amount of work and a consistent maintenance requirement as Netlify iterates on how things work in the first place.

Hope that provides some context!
